server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -193,7 +187,13 @@ public static final class Builder extends FieldMapper.DimensionBuilder {
         private final ScriptCompiler scriptCompiler;
         private final IndexVersion indexCreatedVersion;
 
-        public Builder(String name, IndexAnalyzers indexAnalyzers, ScriptCompiler scriptCompiler, IndexVersion indexCreatedVersion) {
+        public Builder(
+            String name,
+            IndexAnalyzers indexAnalyzers,
+            ScriptCompiler scriptCompiler,
+            int ignoreAboveDefault,
```

Comment:
Unless there are at least a few settings to fetch I would rather keep passing a native integer instead of a reference to an object. A native integer passed on the stack is a bit easier to "clean up" when a method goes out of scope and does not require work from the garbage collector. I would agree with you if we were using more settings...but I don't see any other setting at the moment.

Also for testing purposes is just easier to pass an integer rather than a whole `Settings`.

Reply:
Also this builder is also used by other classes making use of `KeywordFieldMapper` like `PercolatorFieldMapper`. Passing `Setting` here would require changes to propagate settings there which seems unnecessary to me.

Reply:
But then, you need to reference the index param you want to fetch in all interesting call sites - somewhat leaking an internal detail.. I'd also focus on the consistency argument, let's follow existing patterns where possible.

Reply:
The fact that is done in `NumericFieldMapper` does not meat it is consistent...

See: LegacyGeoShapeFieldMapper, BooleanFieldMapper, DateFieldMapper,GeoPointFieldMapper...

If we want to be consistent maybe they should all be like that. My PR touches like 34 files already....I would prefer not to make it like 50 or 70. This "consistency" refactoring could be done in a separate one.

Reply:
Note also that `NumberFieldMapper` just defines an additional constructor that takes a `Settings` and calls

```
public Builder(
            String name,
            NumberType type,
            ScriptCompiler compiler,
            boolean ignoreMalformedByDefault,
            boolean coerceByDefault,
            IndexVersion indexCreatedVersion,
            IndexMode mode
        )
```

Reply:
Moreover, if I replace this call in the `PercolatorFieldMapper`

```
KeywordFieldMapper.Builder queryMetadataFieldBuilder = new KeywordFieldMapper.Builder(
                name,
                Integer.MAX_VALUE,
                indexCreatedVersion
            );
```

with something like

```
KeywordFieldMapper.Builder queryMetadataFieldBuilder = new KeywordFieldMapper.Builder(
                name,
                Settings.builder().build(),
                indexCreatedVersion
            );
```

I would need to do a `getAsInt(String setting, Integer defaultValue)` with `Integer.MAX_VALUE` as `defaultValue`. This is bug-prone...beucase if instead of empty settings we pass (by mistake) settings including `index.mapping.ingore_above` then this parameter would be sued by the percolator too.

Reply:
What about adding a constructor to the `KeywordFieldMapper.Builder` class that just accepts a `MappingParserContext` as parameter? In production code, all current parameter s(except name) come from it.

 This new constructor would delegate to the existing constructor with 4 parameters (we should change its visibility to private, so that constructor with `String, IndexVersion` can still be used). 

Then next time we add a setting we most likely don't need to change the constructor, which results in a much smaller change.